src/integrationTest/java/org/opensearch/security/http/BasicAuthTests.java

```diff
@@ -0,0 +1,146 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+*/
+package org.opensearch.security.http;
+
+import java.util.List;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+import org.apache.http.HttpHeaders;
+import org.hamcrest.Matchers;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.opensearch.test.framework.TestSecurityConfig.AuthcDomain;
+import org.opensearch.test.framework.TestSecurityConfig.User;
+import org.opensearch.test.framework.cluster.ClusterManager;
+import org.opensearch.test.framework.cluster.LocalCluster;
+import org.opensearch.test.framework.cluster.TestRestClient;
+import org.opensearch.test.framework.cluster.TestRestClient.HttpResponse;
+
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.HttpStatus.SC_UNAUTHORIZED;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsStringIgnoringCase;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class BasicAuthTests {
+	static final User TEST_USER = new User("test_user").password("s3cret");
+
+	public static final String CUSTOM_ATTRIBUTE_NAME = "superhero";
+	static final User SUPER_USER = new User("super-user").password("super-password")
+		.attr(CUSTOM_ATTRIBUTE_NAME, true);
+	public static final String NOT_EXISTING_USER = "not-existing-user";
+	public static final String INVALID_PASSWORD = "secret-password";
+
+	public static final AuthcDomain AUTHC_DOMAIN = new AuthcDomain("basic", 0)
+		.challengingAuthenticator("basic").backend("internal");
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder()
+		.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false)
+		.authc(AUTHC_DOMAIN).users(TEST_USER, SUPER_USER).build();
+
+	@Test
+	public void shouldRespondWith401WhenUserDoesNotExist() {
+		try (TestRestClient client = cluster.getRestClient(NOT_EXISTING_USER, INVALID_PASSWORD)) {
```

Comment:
Would all of these tests pass if an exception is thrown in `getRestClient`? I see this pattern used in a number of places in the codebase and I also see one instance where we add:

```
} catch (IOException ioe) {
    fail("unexpected exception")
}
```

See: https://github.com/opensearch-project/security/blob/1904db559960fca9d87e334172017c412f3e3d0e/src/test/java/org/opensearch/security/auditlog/compliance/ComplianceAuditlogTest.java#L234-L242

Reply:
Read up more here https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

If there is an exception it is bubbled up as if the `getRestClient` was executed inside the `try` block, since there is no `catch(Exception e) {}` block in this test case there is nothing to capture.

TLDR: Exceptions work as expected, the test would fail if there was a problem 🥳